api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -432,19 +450,34 @@ public BiometricData saveBiometricsForPatient(Patient patient, BiometricData bio
 				}
 				if (identifierExists) {
 					log.debug("Identifier already exists for patient");
-				}
-				else {
-					PatientIdentifier identifier = new PatientIdentifier(subject.getSubjectId(), idType, null);
+				} else {
+					PatientIdentifier identifier = identifierBuilder.createIdentifier(idType.getUuid(), subject.getSubjectId(), null);
 					patient.addIdentifier(identifier);
 					patientService.savePatientIdentifier(identifier);
 					log.debug("New patient identifier saved for patient: " + identifier);
 				}
-			}
-			else {
+			} else {
 				// TODO: In the future we could add additional support for different storage options - eg. as person attributes
 				throw new IllegalArgumentException("Invalid biometric configuration.  No patient identifier type specified");
 			}
 		}
 		return biometricData;
 	}
+
+	@Override
+	public BiometricData generateBiometricData(BiometricSubject biometricSubject) {
```

Comment:
@dserkowski I am very suspicious about adding new methods to the registration core service. Why do we need these new methods? The generate biometric data methods seems like something that definitely should not be a public method on its own - it should be a private method used by other methods. 

The method overload is also a bit suspicious - why do we need it all of a sudden?